### PR TITLE
CHE-1222: add ability to share workspace with other users

### PR DESCRIPTION
**Pull request by ashumilova**
sharing_workspace

### CODE REVIEW

File: dashboard/src/app/workspace/share-workspace/share-workspace.controller.js [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+      this.isLoading = false;
+      this.cheNotification.showError(error.data && error.data.message ? error.data.message : 'Failed to remove user ' + user.email + ' permissions.');
+    });
+  }
+
+  /**
+   * Find user by its email.
+   *
+   * @param email user's email
+   * @returns {*}
```
Comment (benoitf) [inline dashboard/src/app/workspace/share-workspace/share-workspace.controller.js]: it doesn't seem clear to me that when we find a user, we're returning here the email (and not the 'user' object)

File: dashboard/src/app/workspace/share-workspace/share-workspace.controller.js [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+  getRecipeId() {
+    let workspace = this.cheWorkspace.getWorkspaceById(this.workspaceId);
+    let environments = workspace.config.environments;
+    let defaultEnvName = workspace.config.defaultEnv;
+
+    let defaultEnvironment = this.lodash.find(environments, (environment) => {
+        return environment.name === defaultEnvName;
+    });
+
+    let recipeLocation = defaultEnvironment.machineConfigs[0].source.location;
```
Comment (benoitf) [inline dashboard/src/app/workspace/share-workspace/share-workspace.controller.js]: maybe now we've to use loadash as well to get the machine which is "dev" machine

File: dashboard/src/app/workspace/share-workspace/user-item/user-item.controller.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+   */
+  constructor(codenvyPermissions) {
+    this.codenvyPermissions = codenvyPermissions;
+  }
+
+  /**
+   * Call user permissions removal.
+   */
+  removeUser() {
+    this.callback.removePermissions(this.user);
```
Comment (benoitf) [inline dashboard/src/app/workspace/share-workspace/user-item/user-item.controller.js]: maybe tell that this.callback is coming from the scope definition (as it's not in constructor)

File: dashboard/src/app/workspace/share-workspace/share-workspace.controller.js (hunk 1 above)
Comment (ashumilova) [inline dashboard/src/app/workspace/share-workspace/share-workspace.controller.js]: yes will rename it and rework a little, cause in general its handling email addition and starts requesting the user exists to check

**Pull request merged**